platform/engineering/design-docs/2020-09-24-package-forms-library.md

```diff
@@ -0,0 +1,217 @@
+- Remove forms library dependence on platform re-implementations of 3rd party functions, like lodash's `get` and `set`.
+  - Some of these examples will be easy to avoid the lodash functions entirely, but for others we should switch to the lodash functions instead of the platform versions, with a longer term goal being to get away from lodash altogether
+- Remove imports for `src/platform/utilities/environment` and instead rely on global `const`s from webpack's DefinePlugin for environment-related `const`s (i.e. an api url) or branching
+
+
+#### Final touches
+
+Once the forms library is actually a package, app code can be updated to import from it directly instead of relying on the aliases.
+
+- Each app team will update their code to use forms library JS & SCSS from the package instead of by direct path
+- Configure the `no-restricted-imports` ESLint rule to [restrict imports of `src/platform/packages` directly](https://eslint.org/docs/rules/no-restricted-imports)
+- Remove the `.babelrc` forms-library alias & custom SASS importer
+
+### Code Location
+
+The forms library package will live in `src/platform/packages/forms-library`.
+Since it will be a Node package, it will be added to the `package.json` file and apps will import it as a regular Node package instead of by its direct path.
+
+### Testing Plan
+
+We have ESLint rules in place which can verify that the right JS is imported as it relates to the forms library.
+Additionally, many of the end-to-end tests will fail if the forms library is not imported properly, so they will help with awareness.
+
+We don't have any automation in place to validate that the right SASS files are imported (other than webpack failing to resolve the file), so testing that forms styling is working properly will be a manual process using the browser dev tools.
+
+### Logging
+
+N/A
+
+### Debugging
+
+Since this project isn't about changing functionality of the forms library, we don't need to add any debugging tools. Validating that the app code can still access the forms library in its new location will be enough.
+
+### Caveats
+
+Though we will be creating a package for the forms library, we will not be following semantic versioning since it is _just_ a package and not a published module.
+The additional overhead of updating the forms-library's `package.json` file and updating the `yarn.lock` file in `vets-website` to target the newest (but unpublished) version seems like a burden without much benefit.
+
+### Security Concerns
+
+N/A
+
+### Privacy Concerns
+
+N/A
+
+### Open Questions and Risks
```

Comment:
More questions you can either answer in the document or add to this section:
- Will the forms library be pre-transpiled as it would be if it were published?
  - Will engineers be required to commit `dist/` (or whatever we call the directory containing the transpiled code)?
- Will the forms library manage its own dependencies?
  - How are dependencies of the forms library installed?
  - How do we move dependencies from `vets-website/package.json` to the forms library `package.json` and ensure we're not breaking anything?
  - How will we manage shared dependencies like `React`? (Will we have to?)
- Will the `vets-website` unit test runner also run the forms library unit tests, or will that be a different command and step in CI?